lumicks/pylake/nb_widgets/kymotracker_widgets.py

```diff
@@ -95,6 +105,7 @@ def track_kymo(self, click, release):
         Removes lines in a region, and traces new ones."""
         p1 = [click.xdata, click.ydata]
         p2 = [release.xdata, release.ydata]
+        self.lines = copy(self.lines)  # Explicit copy to maintain history, since we modify in-place
```

Comment:
I would argue that `copy` is needed within `UndoableState` since its purpose is defeated if it doesn't hold separate copies.

Reply:
Yeah, that sounds reasonable. For an immutable object, we could just have a very thin copy constructor.